src/lib/CustomRulePathManager.ts

```diff
@@ -64,6 +64,11 @@ export class CustomRulePathManager {
     return await this.saveCustomClasspaths();
   }
 
+  public async getRulePathEntries(engine: ENGINE): Promise<Map<string, Set<string>>> {
```

Comment:
Nitpick: `getRulePathEntriesForEngine()`?

Reply:
This is an API convention where Engine is implied when we pass it as parameter.